apps/ucp-ui/src/Institutions/Institutions.tsx

```diff
@@ -1,40 +1,137 @@
 import {
   Avatar,
   Chip,
+  Pagination,
   Table,
   TableBody,
   TableCell,
   TableContainer,
   TableHead,
+  TablePagination,
   TableRow,
+  Tooltip,
 } from "@mui/material";
 import React from "react";
 import PageTitle from "../shared/components/PageTitle";
 import {
+  INSTITUTIONS_AGGREGATOR_INFO_ICON,
+  INSTITUTIONS_AGGREGATOR_INFO_TOOLTIP,
+  INSTITUTIONS_ERROR_TEXT,
   INSTITUTIONS_PAGE_TITLE,
   INSTITUTIONS_PERMISSIONS_ERROR_TEXT,
   INSTITUTIONS_ROW_TEST_ID,
   INSTITUTITIONS_ROW_AGGREGATOR_CHIP_TEST_ID,
 } from "./constants";
 import styles from "./institutions.module.css";
-import { institutions } from "./testData/institutions";
 import AddInstitution from "./ChangeInstitution/AddInstitution";
 import FetchError from "../shared/components/FetchError";
-import { useGetInstitutionPermissionsQuery } from "./api";
+import {
+  AggregatorIntegration,
+  useGetInstitutionPermissionsQuery,
+  useGetInstitutionsQuery,
+} from "./api";
 import PageContent from "../shared/components/PageContent";
+import { supportsJobTypeMap } from "../shared/constants/jobTypes";
+import { InfoOutlined } from "@mui/icons-material";
+import { useSearchParams } from "react-router-dom";
+import {
+  SkeletonIfLoading,
+  TextSkeletonIfLoading,
+} from "../shared/components/Skeleton";
+
+const generateFakeInstitutionData = (rowsPerPage: number) => {
+  return new Array(rowsPerPage).fill(0).map(() => ({
+    id: window.crypto.randomUUID(),
+    logo: undefined,
+    name: "Test that has to be quite long 1234566777",
+    aggregatorIntegrations: new Array(Math.floor(Math.random() * 2) + 1)
+      .fill(0)
+      .map((_, index) => ({
+        id: index,
+        isActive: true,
+        ...Object.values(supportsJobTypeMap).reduce(
+          (acc, { prop }) => ({
+            ...acc,
+            [prop]: true,
+          }),
+          {},
+        ),
+        aggregator: {
+          displayName: (Math.random() + 1).toString(36).substring(7),
+        },
+      })),
+  }));
+};
```

Comment:
Is this method so the skeletons work prior to data being available?

Reply:
Yeah i'm just adding some variance so it looks somewhat real